src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -186,7 +296,12 @@ public void OnHttpRequestInStart(HttpContext httpContext)
                 }
 
                 requestTelemetry.Context.Operation.ParentId = originalParentId;
-                SetAppIdInResponseHeader(httpContext, requestTelemetry);
+
+                // Only reply back with AppId if we got an indication that we need to set one
+                if (!string.IsNullOrWhiteSpace(requestTelemetry.Source))
```

Comment:
nit: why it is important to check for whitespace? if someone sent `request-context:  ` I guess it's ok to still reply back with request-context.

Reply:
Makes sense, I'll execute null-check only.